pkg/pillar/zedUpload/datastore_aws.go

```diff
@@ -193,6 +196,30 @@ func (ep *AwsTransportMethod) processS3Download(req *DronaRequest) (error, int)
 	return err, csize
 }
 
+func (ep *AwsTransportMethod) processS3DownloadByChunks(req *DronaRequest) error {
+	pwd := strings.TrimSuffix(ep.apiKey, "\n")
+	dataChan := make(utils.ChunkDataChan)
+	done := make(chan bool)
+	go func(req *DronaRequest, dataNotif utils.ChunkDataChan, donePost chan bool) {
+		for {
+			data, ok := <-dataNotif
+			if !ok {
+				return
+			}
```

Comment:
I've commented twice already that this is better expressed as a for range dataNotify.
Did you not see those comments?

Reply:
i tested the code replaced with  range dataNotify, it's trying to post when dataNotif is closed

Reply:
let me test it end to end once again and update.

Reply:
sorry! my bad, I over thinked about channel sending empty data upon closing. 
updated as you suggested.